tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py

```diff
@@ -218,6 +229,168 @@ def test_container_combos_invalid(
                 code_section,
                 first_sub_container,
             ],
+            kind=container_kind,
+        ),
+        expect_exception=EOFException.INCOMPATIBLE_CONTAINER_KIND,
+    )
+
+
+@pytest.mark.parametrize(
+    "code_section,first_sub_container",
+    [
+        pytest.param(
+            eofcreate_revert_code_section,
+            returncontract_sub_container,
+            id="EOFCREATE/RETURNCONTRACT",
+        ),
+        pytest.param(
+            returncontract_code_section,
+            stop_sub_container,
+            id="RETURNCONTRACT/STOP",
+        ),
+        pytest.param(
+            returncontract_code_section,
+            return_sub_container,
+            id="RETURNCONTRACT/RETURN",
+        ),
+        pytest.param(
+            eofcreate_revert_code_section,
+            revert_sub_container,
+            id="EOFCREATE/REVERT",
+        ),
+        pytest.param(
+            returncontract_code_section,
+            revert_sub_container,
+            id="RETURNCONTRACT/REVERT",
+        ),
+    ],
+)
+def test_container_combos_deeply_nested_valid(
+    eof_test: EOFTestFiller,
+    code_section: Section,
+    first_sub_container: Container,
+):
+    """Test valid subcontainer reference / opcode combos on a deep container nesting level"""
+    valid_container = Container(
+        sections=[
+            code_section,
+            first_sub_container,
+        ],
+        kind=ContainerKind.INITCODE,
+    )
+
+    container = valid_container
+    while len(container) < MAX_BYTECODE_SIZE:
+        container_copy = container.copy()
+
+        # container.copy() sets all fields and sections' custom_size becomes 0
```

Comment:
So in the container produced by `copy()` all fields are present in `container.model_fields_set`, then bytecode is produced it thinks that `custom_size` is set to `0`, and puts 0 section size into header.

I tried to fix `copy()` by filtering out the fields not in source `model_fields_set`, but this didn't work so easy, because it's a hierarchy of objects, it needs to do the filtering when doing recursive `copy()` of nested `Section` objects...
@marioevz any other thoughts how to address this?

Reply:
Fixed in #679